### PR TITLE
fix: subsequent sub model split #154

### PR DESCRIPTION
**Pull request by leninmehedy**
## Types of changes

- \[x\] Bug fix (non-breaking change which fixes an issue)
- \[x\] New feature (non-breaking change which adds functionality)
- \[x\] Breaking change (fix or feature that would cause existing functionality to change)
- \[x\] My code follows the code style of this project.
- \[x\] My change requires a change to the documentation.
- \[ \] I have updated the documentation accordingly.
- \[x\] I have added tests to cover my changes.
- \[x\] All new and existing tests passed.

It fix the issues #154 and #148

### CODE REVIEW

**Comment by leninmehedy** [comment]
split command output:
```
(venv) ➜  mycatalog git:(fix/sub_model_split_154) tsl split -f catalog.json -e 'catalog.groups.*,catalog.metadata.parties.*'
(venv) ➜  mycatalog git:(fix/sub_model_split_154) tree
.
├── catalog
│   ├── groups
│   │   ├── 00000__group.json
│   │   ├── 00001__group.json
│   │   ├── 00002__group.json
│   │   ├── 00003__group.json
│   │   ├── 00004__group.json
│   │   ├── 00005__group.json
│   │   ├── 00006__group.json
│   │   ├── 00007__group.json
│   │   ├── 00008__group.json
│   │   ├── 00009__group.json
│   │   ├── 00010__group.json
│   │   ├── 00011__group.json
│   │   ├── 00012__group.json
│   │   ├── 00013__group.json
│   │   ├── 00014__group.json
│   │   ├── 00015__group.json
│   │   ├── 00016__group.json
│   │   ├── 00017__group.json
│   │   ├── 00018__group.json
│   │   └── 00019__group.json
│   └── metadata
│       └── parties
│           └── 00000__party.json
└── catalog.json

4 directories, 22 files
```

File: tests/trestle/core/commands/cmd_utils_test.py [hunk 1]
```diff
@@ -38,58 +39,70 @@ def prepare_expected_element_paths(element_args: List[str]) -> List[ElementPath]
...
+    metadata_values = sample_target_def.metadata.__dict__
     metadata_values['title'] = 'TEST'
     sample_metadata2: target.Metadata = target.Metadata(**metadata_values)
-    assert sample_metadata2 is not sample_target.metadata
+    assert sample_metadata2 is not sample_target_def.metadata
 
-    cmd_utils.copy_values(sample_target.metadata, sample_metadata2)
-    assert sample_metadata2 == sample_target.metadata
+    cmd_utils.copy_values(sample_target_def.metadata, sample_metadata2)
+    assert sample_metadata2 == sample_target_def.metadata
```
Comment (butler54) [inline tests/trestle/core/commands/cmd_utils_test.py:50]: this functionality already exists in copy_from in the oscal base model from my understanding. This seems duplicative. Surely we can compose ontop of the original functions.

Comment (butler54) [reply tests/trestle/core/commands/cmd_utils_test.py:50]: Correction - this function is not necessary. We should be using https://pydantic-docs.helpmanual.io/usage/exporting_models/#modelcopy

File: tests/trestle/core/models/element_test.py [hunk 2]
```diff
@@ -24,32 +24,32 @@
...
-    assert element.get_at(ElementPath('target-definition.metadata.parties.0')) == sample_target.metadata.parties[0]
+    assert element.get_at(ElementPath('target-definition.metadata')) == sample_target_def.metadata
+    assert element.get_at(ElementPath('target-definition.metadata.title')) == sample_target_def.metadata.title
+    assert element.get_at(ElementPath('target-definition.targets')) == sample_target_def.targets
+    assert element.get_at(ElementPath('target-definition.targets.*')) == sample_target_def.targets
+    assert element.get_at(ElementPath('target-definition.metadata.parties.*')) == sample_target_def.metadata.parties
+    assert element.get_at(ElementPath('target-definition.metadata.parties.0')) == sample_target_def.metadata.parties[0]
     assert element.get_at(ElementPath('target-definition.metadata.parties.0.uuid')
-                          ) == sample_target.metadata.parties[0].uuid
+                          ) == sample_target_def.metadata.parties[0].uuid
```
Comment (butler54) [inline tests/trestle/core/models/element_test.py:48]: Going forward I would suggest that we use the 800-53 base catalog as a reference object of OSCAL. The reason why I specifically suggest this is that the 800-53 object provided by NIST they will be responsible for updating to reflect the latest object model. Target-definition is something we own and need to manage updates for.

Comment (butler54) [reply tests/trestle/core/models/element_test.py:48]: This is not a blocker but a strong recommendtaion going foward.

**Review by butler54 — APPROVED** [review]
LGTM - But see the issues on copy it seems like we have redundant functionality to pydantic copy.

File: tests/trestle/core/commands/cmd_utils_test.py (hunk 1 above)
Comment (leninmehedy) [reply tests/trestle/core/commands/cmd_utils_test.py:50]: Seems this is an unused method. So we can just delete this.

File: tests/trestle/core/models/element_test.py (hunk 2 above)
Comment (leninmehedy) [reply tests/trestle/core/models/element_test.py:48]: ok. Thanks.

**Pull request merged**